cmd/queue/main.go

```diff
@@ -187,18 +200,26 @@ func handler(reqChan chan queue.ReqEvent, breaker *queue.Breaker, handler http.H
 					w.Write([]byte(queue.Name))
 				} else {
 					http.Error(w, "container not ready", http.StatusServiceUnavailable)
+					probeSpan.Annotate([]trace.Attribute{
+						trace.StringAttribute("queueproxy.probe.error", "container not ready")}, "error")
 				}
+
 			} else {
 				http.Error(w, "no probe", http.StatusInternalServerError)
+				probeSpan.Annotate([]trace.Attribute{
+					trace.StringAttribute("queueproxy.probe.error", "no probe")}, "error")
 			}
-
+			probeSpan.End()
 			return
 		case network.IsKubeletProbe(r):
 			// Do not count health checks for concurrency metrics
-			handler.ServeHTTP(w, r)
+			handler.ServeHTTP(w, r.WithContext(probeCtx))
+			probeSpan.End()
 			return
 		}
-
+		probeSpan.End()
```

Comment:
Is this probeSpan.End() redudant now?

Reply:
It's not redundant in the scenario that neither `case` in the switch statement is hit.

Reply:
That's... a very good point! Should we have that span at all in such a case?

Reply:
🤔I think in that case, there is no actual probe happening, although I guess there's a tiny bit of work associated with the probe (checking the header). 

I haven't had to debug anything on queue-proxy, but I can imagine that baseline span might be a nice to have for comparison?

Reply:
The probeSpan would just be a noop though, right? As in: Barely even there.

Reply:
Yes, correct - in that case, it's only checking the header & then moving on.  Let me take a look at reworking where we start the span.